pkg/tfgen/docs.go

```diff
@@ -456,29 +531,41 @@ func splitGroupLines(s, sep string) [][]string {
 // parseTFMarkdown takes a TF website markdown doc and extracts a structured representation for use in
 // generating doc comments
 func parseTFMarkdown(g *Generator, info tfbridge.ResourceOrDataSourceInfo, kind DocKind,
-	markdown, markdownFileName, resourcePrefix, rawname string) (entityDocs, error) {
+	markdown []byte, markdownFileName, resourcePrefix, rawname string) (entityDocs, error) {
 
 	p := &tfMarkdownParser{
-		g:                g,
+		sink:             g,
 		info:             info,
 		kind:             kind,
-		markdown:         markdown,
 		markdownFileName: markdownFileName,
-		resourcePrefix:   resourcePrefix,
 		rawname:          rawname,
+
+		infoCtx: infoContext{
+			language: g.language,
+			pkg:      g.pkg,
+			info:     g.info,
+		},
+		docRules: g.info.DocRules,
 	}
-	return p.parse()
+	return p.parse(markdown)
```

Comment:
Nit: does the functionality of replacing the markdown belong in a structure called 'markdownParser' or can it be easily expressed as a transformation on the input that happens before its passed to the parser? I guess which one's easier to test?

Reply:
That makes sense then! Ty.